html/components/data-management-tab.js

```diff
@@ -0,0 +1,249 @@
+		var buttonsCloseOnly = { Close: closeFunction };
+		var buttonsUploadClose = { Upload: uploadFunction, Close: closeFunction };
+
+		div.dialog({
+			title: "Upload media "+format+" : "+type,
+			modal: true,
+			width: 700,
+			close: function() { $(this).dialog("destroy").remove(); },
+			buttons: buttonsCloseOnly
+		});
+		inputFile.change(function() {
+			var files = this.files;
+			if (!files || !files.length) {
+				div.dialog("option", "buttons", buttonsCloseOnly);
+				return;
+			}
+			div.dialog("option", "buttons", buttonsUploadClose);
+		}).change();
+	}
+	
+	function createGameDataTab(tab) {
+		var iframe = $('<iframe src="/game-data">').attr('id', 'GameData');
+		tab.append($('<table>')
+				.append($('<td>').addClass('Info').text('To download, right-click and Save - to view JSON, left-click'))
+				.append($('<td>').addClass('Reload').append($('<button>').text('Reload').button().click(function() {
+					iframe.attr('src', '/game-data');
+				}))))
+			.append(iframe);
+		
+		
+	}
+
+	function createSaveLoadTab(tab) {
+		var table = $("<table>").attr("id", "SaveLoad").appendTo(tab);
+		
+		// Download table
+		var sbDownloadTable = $("<table>").addClass("Download")
+			.appendTo($("<td>").appendTo($("<tr>").appendTo(table)));
+		$("<tr>").addClass("Name").appendTo(sbDownloadTable)
+			.append("<td colspan='4'>Download ScoreBoard JSON</td>");
+		$("<tr>").addClass("Instruction").appendTo(sbDownloadTable)
+			.append("<td colspan='4'>To download, right-click and Save - to view JSON, left-click</td>");
+		var contentRow = $("<tr>").addClass("Content").appendTo(sbDownloadTable);
+	
+		var links = [
+		{ name: "All data", url: "" },
+		{ name: "Teams", url: "teams.json?path=ScoreBoard.PreparedTeam" }
+		];
+		$.each( links, function() {
+			$("<td><a/></td>").appendTo(contentRow)
+				.children("a").html(this.name)
+				.attr({ href: "/SaveJSON/"+this.url, target: "_blank" });
+		});
+		var allDataA = contentRow.find(">td:eq(0)>a");
+		var updateAllUrl = function() {
+			var d = new Date();
+			var name = $.datepicker.formatDate("yy-mm-dd_", d);
+			name += _timeConversions.twoDigit(d.getHours());
+			name += _timeConversions.twoDigit(d.getMinutes());
+			name += _timeConversions.twoDigit(d.getSeconds());
+			allDataA.attr("href", "/SaveJSON/scoreboard-"+name+".json");
+		};
+		setInterval(updateAllUrl, 1000);
+	
+	
+		// Upload table
+		var sbUploadTable = $("<table>").addClass("Upload")
+			.appendTo($("<td>").appendTo($("<tr>").appendTo(table)));
+		$("<tr>").addClass("Name").appendTo(sbUploadTable)
+			.append("<td>Upload ScoreBoard JSON</td>");
+		var contentTd = $("<td>")
+			.appendTo($("<tr>").addClass("Content").appendTo(sbUploadTable));
+	
+		var iframeId = "SaveLoadUploadHiddenIframe";
+		var uploadForm = $("<form method='post' enctype='multipart/form-data' target='"+iframeId+"'/>")
+			.append("<iframe id='"+iframeId+"' name='"+iframeId+"' style='display: none'/>")
```

Comment:
I'd prefer that we use completely separate pages and use browser tabs as you previously suggested. This means that refresh keeps you on the same page, and avoids two layers of tabs when looking at media. Iframes are generally something to avoid (and may break with future browser security changes) - plus it makes page loads slower.

Reply:
Using browser tabs has downsides, that I didn't immediately see:
- If you open all of the tabs at once, the user will have to close multiple tabs despite possibly only using one of them.
- If you open the tabs on demand, their order will vary. Also users might see the links that open the tabs as being the default way to switch, but they will trigger a reload, which is undesirable.
- If you run the page in full screen, you won't see the tab bar.

I see the iframe as a temporary solution until someone implements a nicer interface.

Reply:
> Also users might see the links that open the tabs as being the default way to switch, but they will trigger a reload, which is undesirable.

The `target` attribute can handle that.

I'm not worried about full screen for any of these, and if we have a consistent ordering in the list then users will generally open them in that order.

Reply:
`target` will ensure that we don't open multiple tabs for the same screen but it will still reload the tab every time the link is clicked.

And I'm pretty sure most users will by default only open the main operator screen, opening other ones as they need them.

Reply:
This leaves all the problems that the existing system has, actual tabs would make more sense.

Reply:
> all the problems that the existing system has

You named refresh keeping you on the same page, the two layers of tabs and the iframe. - The iframe has since gone. For the two layers of tabs I I think the different colors work really well. This leaves just the refresh. But with the other tabs being available as standalone screens, it's likely that these tabs on the operator screen will only be used for quick changes just before or during the game or downloading the game data afterwards. In all of these cases, the user doesn't want to be on a non-operator tab for a prolonged time, so the chance of a reload taking them  to the 'wrong' tab is pretty slim. Plus, if this is actually a significant problem, it's rather easy to store the current tab as a page parameter and read it on reload (like we do with the operator name).

In my opinion the problems I mentioned above for the browser tab solution are way more serious.

Reply:
> In all of these cases, the user doesn't want to be on a non-operator tab for a prolonged time, so the chance of a reload taking them to the 'wrong' tab is pretty slim.

It depends on what they're doing, if everything is on the operator screen would everyone realise these other screens exist? I still think actual separate pages makes more sense, and limit future issues.